File: leetcode-14-Longest_Common_Prefix/leetcode-14-Longest_Common_Prefix.cpp
#include <iostream>
#include <cstdlib>
#include <string>
#include <vector>


std::string longestCommonPrefix(std::vector<std::string>& strs) {
    std::string common;
    int min_len = strs[0].size();
    
    for(const auto& x:strs){
        if(x.size()<min_len){
            min_len = x.size();
        }
    }

    for(size_t i = 0; i<min_len; ++i){
        char c = strs[0][i];
        for(size_t j = 0; j<strs.size(); ++j){
            if(strs[j][i] != c)
            {
                return common;
            }
        }
        common += c;
    }
    return common;
}


int main(){
    std::vector<std::string> vec = {"flower","flow","flight"};
    
    auto res = longestCommonPrefix(vec);
    
    std::cout << res << std::endl;
    
    return EXIT_SUCCESS;
}

File: leetcode-14-Longest_Common_Prefix/leetcode-14-Longest_Common_Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        result: str = ""
        min_len: int = len(min(strs, key=len))
        for i in range(min_len):
            temp_char: str = ""
            for s in strs:
                if temp_char == "":
                    temp_char = s[i]
                elif temp_char != s[i]:
                    return result
            result += temp_char
        return result


if __name__ == "__main__":
    strs: list[str] = ["flower","flow","flight"]
    result = Solution().longestCommonPrefix(strs)
    print(result)
